Applications/Spire/Source/Ui/InfoPanel.cpp

```diff
@@ -0,0 +1,151 @@
+#include "Spire/Ui/InfoPanel.hpp"
+#include <QEvent>
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Ui/Button.hpp"
+#include "Spire/Ui/Icon.hpp"
+#include "Spire/Ui/Layouts.hpp"
+#include "Spire/Ui/OverlayPanel.hpp"
+#include "Spire/Ui/TextAreaBox.hpp"
+#include "Spire/Ui/TextBox.hpp"
+
+using namespace boost;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto SEVERITY_ICON_SIZE() {
+    static auto size = scale(14, 14);
+    return size;
+  }
+
+  const auto& INFO_IMAGE() {
+    static auto image =
+      imageFromSvg(":/Icons/info_panel/info.svg", SEVERITY_ICON_SIZE());
+    return image;
+  }
+
+  const auto& SUCCESS_IMAGE() {
+    static auto image =
+      imageFromSvg(":/Icons/info_panel/success.svg", SEVERITY_ICON_SIZE());
+    return image;
+  }
+
+  const auto& WARNING_IMAGE() {
+    static auto image =
+      imageFromSvg(":/Icons/info_panel/warning.svg", SEVERITY_ICON_SIZE());
+    return image;
+  }
+
+  const auto& ERROR_IMAGE() {
+    static auto image =
+      imageFromSvg(":/Icons/info_panel/error.svg", SEVERITY_ICON_SIZE());
+    return image;
+  }
+}
+
+InfoPanel::InfoPanel(QString message, QWidget& parent)
+  : InfoPanel(Severity::INFO, std::move(message), parent) {}
+
+InfoPanel::InfoPanel(Severity severity, QString message, QWidget& parent)
+    : QWidget(&parent),
+      m_severity(severity),
+      m_message(std::move(message)) {
+  setAttribute(Qt::WA_DeleteOnClose);
+  auto image = [&] {
+    if(m_severity == Severity::INFO) {
+      return INFO_IMAGE();
+    } else if(m_severity == Severity::SUCCESS) {
+      return SUCCESS_IMAGE();
+    } else if(m_severity == Severity::WARNING) {
+      return WARNING_IMAGE();
+    } else if(m_severity == Severity::ERROR) {
+      return ERROR_IMAGE();
+    }
+    return QImage();
+  }();
+  auto severity_icon = new Icon(image);
+  severity_icon->setFixedSize(SEVERITY_ICON_SIZE());
+  update_style(*severity_icon, [&] (auto& styles) {
+    if(m_severity == Severity::INFO) {
+      styles.get(Any() || Hover()).set(Fill(QColor(0x4392D6)));
```

Comment:
Applying styles to Any() can't override the default Hover styles.

Reply:
If you're looking to set the style instead of extend it, then use set_style, along the lines of:
    auto style = StyleSheet();
    if(m_severity == Severity::INFO) {
      style.get(Any()).set(...);
    } else if(...) {
    }
    ...
    set_style(*severity_icon, std::move(style));